### PR TITLE
Add ShowChan & minimize ListChans columns

### PR DESCRIPTION
**Pull request by jpnurmi**

### CODE REVIEW

**Comment by jpnurmi** [comment]
How about something like this?

**Comment by Mikaela** [comment]
How does this look like to users?

**Comment by jpnurmi** [comment]
This won't remove the table (yet), just reducing the amount of columns to make it possible later.
```
<jpnurmi> listchans
<*status> +=========+========+
<*status> | Name    | Status |
<*status> +=========+========+
<*status> | #foobar | Joined |
<*status> +=========+========+
<*status> Total: 1 - Joined: 1 - Detached: 0 - Disabled: 0

<jp-znc> showchan #foobar
<*status> +===================+====================================+
<*status> | #foobar           | Joined                             |
<*status> +===================+====================================+
<*status> | Topic             | This can get long and thus wrap... |
<*status> +-------------------+------------------------------------+
<*status> | Configured        | yes                                |
<*status> +-------------------+------------------------------------+
<*status> | Buffer size       | 500 (default)                      |
<*status> +-------------------+------------------------------------+
<*status> | Auto clear buffer | no (default)                       |
<*status> +-------------------+------------------------------------+
<*status> | Modes             | +nst                               |
<*status> +-------------------+------------------------------------+
<*status> | Total users       | 123                                |
<*status> +-------------------+------------------------------------+
<*status> | @'d users         | 1                                  |
<*status> +-------------------+------------------------------------+
<*status> | +'d users         | 2                                  |
<*status> +===================+====================================+
```
Topic can be removed or truncated/ellipsized if desired...

**Comment by jpnurmi** [comment]
Revised output:
```
<jpnurmi> showchan #foobar
<*status> +=====================+====================================+
<*status> | #foobar             | Joined                             |
<*status> +=====================+====================================+
<*status> | InConfig            | yes                                |
<*status> +---------------------+------------------------------------+
<*status> | Buffer              | 6/1500 (default)                   |
<*status> +---------------------+------------------------------------+
<*status> | AutoClearChanBuffer | no (default)                       |
<*status> +---------------------+------------------------------------+
<*status> | Topic               | This can get long and thus wrap... |
<*status> +---------------------+------------------------------------+
<*status> | Modes               | +nst                               |
<*status> +---------------------+------------------------------------+
<*status> | Users               | All: 123, @: 1, +: 2               |
<*status> +=====================+====================================+
```
- now you can see the used buffer size
- users, ops, voices merged to a single line to keep the output compact
- the last three lines are only shown when the status is "joined"

**Comment by jpnurmi** [comment]
Actually Buffer & AutoClearBuffer rows could be also merged:
```
<*status> +----------+-----------------------------------------+
<*status> | Buffer   | 123/500 (default), auto clear (default) |
<*status> +----------+-----------------------------------------+
```
Any comments?

File: src/ClientCommand.cpp [hunk 1]
```diff
@@ -423,6 +423,67 @@ void CClient::UserCommand(CString& sLine) {
...
+			return;
+		}
+
+		CChan* pChan = m_pNetwork->FindChan(sChan);
+		if (!pChan) {
+			PutStatus("No such channel [" + sChan + "]");
+			return;
+		}
+		sChan = pChan->GetPermStr() + pChan->GetName();
+		CString sStatus = pChan->IsOn() ? pChan->IsDetached() ? "Detached" : "Joined" : pChan->IsDisabled() ? "Disabled" : "Trying";
```
Comment (DarthGandalf) [inline src/ClientCommand.cpp]: Please add `(` `)`

File: src/ClientCommand.cpp [hunk 2]
```diff
@@ -423,6 +423,67 @@ void CClient::UserCommand(CString& sLine) {
...
+		Table.SetCell(sChan, "InConfig");
+		Table.SetCell(sStatus, CString((pChan->InConfig()) ? "yes" : "no"));
+
+		Table.AddRow();
+		Table.SetCell(sChan, "Buffer");
+		Table.SetCell(sStatus, CString(pChan->GetBuffer().Size()) + "/" + CString(pChan->GetBufferCount()) + CString(pChan->HasBufferCountSet() ? "" : " (default)"));
+
+		Table.AddRow();
+		Table.SetCell(sChan, "AutoClearChanBuffer");
+		Table.SetCell(sStatus, CString(CString(pChan->AutoClearChanBuffer() ? "yes" : "no") + CString(pChan->HasAutoClearChanBufferSet() ? "" : " (default)")));
```
Comment (DarthGandalf) [inline src/ClientCommand.cpp]: `CString(CString()+CString())`?

File: src/ClientCommand.cpp (hunk 1 above)
Comment (jpnurmi) [inline src/ClientCommand.cpp]: Done

File: src/ClientCommand.cpp (hunk 2 above)
Comment (jpnurmi) [inline src/ClientCommand.cpp]: Done

File: src/ClientCommand.cpp [hunk 3]
```diff
@@ -423,6 +423,67 @@ void CClient::UserCommand(CString& sLine) {
...
+		Table.SetCell(sChan, "InConfig");
+		Table.SetCell(sStatus, CString(pChan->InConfig() ? "yes" : "no"));
+
+		Table.AddRow();
+		Table.SetCell(sChan, "Buffer");
+		Table.SetCell(sStatus, CString(pChan->GetBuffer().Size()) + "/" + CString(pChan->GetBufferCount()) + CString(pChan->HasBufferCountSet() ? "" : " (default)"));
+
+		Table.AddRow();
+		Table.SetCell(sChan, "AutoClearChanBuffer");
+		Table.SetCell(sStatus, CString(pChan->AutoClearChanBuffer() ? "yes" : "no") + CString(pChan->HasAutoClearChanBufferSet() ? "" : " (default)"));
```
Comment (DarthGandalf) [inline src/ClientCommand.cpp]: With C++14, it will be possible to use `"yes"s` instead of `CString()`

**Pull request merged**